src/arch/riscv/init.c

```diff
@@ -174,7 +174,7 @@ int start_secondary(struct sys_info *sys) {
 
 void my_monitor_entry(void);
 
-void init_full(unsigned long hartid, unsigned long fdt) {
```

Comment:
Might make sense to have a "simple init" behind a Kconfig flag. Would save us from modifying init to change how the system boots.

Further, it might be time to look into platform specific boot routines a-la the x86 KNL impl. Idk, we might eventually wanna run nautilus on the little hifive embedded board in m-mode :)